lib/ansible/modules/crypto/openssl_pkcs12.py

```diff
@@ -0,0 +1,365 @@
+    certificate_path: '/opt/certs/cert.pem'
+    ca_certificates: '/opt/certs/ca.pem'
+    state: present
+    mode: 0600
+    force: True
+
+- name: 'Dump/Parse PKCS#12 file'
+  openssl_pkcs12:
+    action: parse
+    src: '/opt/certs/ansible.p12'
+    path: '/opt/certs/ansible.pem'
+    state: present
+
+- name: 'Remove PKCS#12 file'
+  openssl_pkcs12:
+    path: '/opt/certs/ansible.p12'
+    state: absent
+'''
+
+RETURN = '''
+filename:
+    description: Path to the generate PKCS#12 file.
+    returned: changed or success
+    type: string
+    sample: /opt/certs/ansible.p12
+privatekey:
+    description: Path to the TLS/SSL private key the public key was generated from
+    returned: changed or success
+    type: string
+    sample: /etc/ssl/private/ansible.com.pem
+'''
+
+import os
+
+try:
+    from OpenSSL import crypto
+except ImportError:
+    pyopenssl_found = False
+else:
+    pyopenssl_found = True
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils import crypto as crypto_utils
+from ansible.module_utils._text import to_native
+
+
+class PkcsError(crypto_utils.OpenSSLObjectError):
+    pass
+
+
+class Pkcs(crypto_utils.OpenSSLObject):
+
+    def __init__(self, module):
+        super(Pkcs, self).__init__(
+            module.params['path'],
+            module.params['force'],
+            module.params['state'],
+            module.check_mode
+        )
+        self.action = module.params['action']
+        self.ca_certificates = module.params['ca_certificates']
+        self.certificate_path = module.params['certificate_path']
+        self.friendly_name = module.params['friendly_name']
+        self.iter_size = module.params['iter_size']
+        self.maciter_size = module.params['maciter_size']
+        self.passphrase = module.params['passphrase']
+        self.pkcs12 = None
+        self.privatekey_passphrase = module.params['privatekey_passphrase']
+        self.privatekey_path = module.params['privatekey_path']
+        self.src = module.params['src']
+        self.mode = module.params['mode']
+        if not self.mode:
+            self.mode = int('0400', 8)
+
+    def check(self, module, perms_required=True):
+        """Ensure the resource is in its desired state."""
+
+        state_and_perms = super(Pkcs, self).check(module, perms_required)
+
+        def _check_pkey_passphrase():
+            if self.privatekey_passphrase:
+                try:
+                    crypto_utils.load_privatekey(self.path,
+                                                 self.privatekey_passphrase)
+                    return True
+                except crypto.Error:
+                    return False
+            return True
+
+        if not state_and_perms:
+            return state_and_perms
+
+        return _check_pkey_passphrase
+
+    def dump(self):
+        """Serialize the object into a dictionary."""
+
+        result = {
+            'changed': self.changed,
```

Comment:
Not sure why changed is added here, it could make sense to set value of `changed` key [here](https://github.com/ansible/ansible/pull/27320/files#diff-15692372a20724add1fc115ef36deff9R360).